### ISSUE TITLE
Add default postcode comparison function

### ISSUE THREAD

**Issue by samnlindsay** [moj-analytical-services/splink#215]
![image](https://user-images.githubusercontent.com/7570107/136946496-8769b06c-e4a6-488d-95f1-526946d96aa7.png)

Generate a case expression with 3-5 levels:

```
CASE 
WHEN {full_match} THEN 4
WHEN {sector_match} THEN 3
WHEN {district_match} THEN 2
WHEN {area_match} THEN 1
ELSE 0 
END
```

**Comment by RobinL** [comment]
Yeah, needs to be in the form of a SQL case expression

**Comment by mamonu** [comment]
have a look on Slack for a ScalaUDF solution. Or is there an easier way and I am overcomplicating things?

**Comment by pbhj** [comment]
Is that too naive? Wouldn't it be better to convert to a more general geographic indicator (rather than a postal area), like lat/lon, and then compare? With this sort of comparison you might have moved one street but have an entirely different postcode?

There must be a lot of solutions to this already with UK gov codebases.

**Comment by RobinL** [comment]
We have support for distance as the crow flies (not yet in the pypi version, will land in the next release):
https://github.com/moj-analytical-services/splink/blob/d80b0c72a71ebb93dfb2aef76f5f1ceb458f7623/splink/comparison_level_library.py#L228

It can just sometimes be a bit of a faff to join on the geolocation, so this would provide a quickwand dirty solution that would get 80% of the way there

**Comment by RossKen** [comment]
Could build ontop of #1190

**Comment by RossKen** [comment]
Closed by #1230
